CMakeLists.txt

```diff
@@ -107,103 +91,51 @@ if(ENABLE_CCACHE AND (CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILE
   endif()
 endif()
 
-include(cmake/SanitizerOptions.cmake)
-
-## Coverage report targets
-if(ENABLE_COVERAGE)
-  find_program(GENHTML_PATH NAMES genhtml genhtml.perl genhtml.bat)
-  if(NOT GENHTML_PATH)
-    message(FATAL_ERROR "no genhtml installed")
-  endif()
-
-  set(FASTCOV_PATH ${VALHALLA_SOURCE_DIR}/third_party/fastcov/fastcov.py)
-  add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/coverage.info
-    COMMAND ${FASTCOV_PATH} -d . --exclude /usr/ third_party/ ${CMAKE_CURRENT_BINARY_DIR}/ --lcov -o coverage.info
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-    DEPENDS check)
-
-  add_custom_target(coverage
-    COMMAND ${GENHTML_PATH} --prefix ${CMAKE_CURRENT_BINARY_DIR} --output-directory coverage --title "Test Coverage" --legend --show-details coverage.info
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/coverage.info)
-  set_target_properties(coverage PROPERTIES FOLDER "Tests")
-endif()
+include(ValhallaSanitizerOptions)
+include(ValhallaSourceGroups)
 
 ## Dependencies
 find_package(Threads REQUIRED)
 find_package(ZLIB REQUIRED)
 
-# try to find an installed boost or install locally with conan
-set(boost_VERSION "1.71")
-find_package(Boost ${boost_VERSION} QUIET)
-if (NOT Boost_FOUND)
-  # bail if there's no conan installed
-  message(STATUS "No compatible boost version detected, using conan...")
-  find_program(conan_FOUND conan)
-  if (NOT conan_FOUND)
-    message(FATAL_ERROR "conan needs to be installed for boost, see https://docs.conan.io/en/latest/installation.html.")
+# cURL
+set(curl_targets "")
+if (ENABLE_HTTP OR ENABLE_DATA_TOOLS)
+  find_package(CURL)
+  if (CURL_FOUND)
+    set(curl_targets CURL::libcurl)
+    target_compile_definitions(CURL::libcurl INTERFACE ENABLE_HTTP)
+    if (MSVC)
+      # TODO(nils): this seems too blankety.. in CI python_valhalla fails
+      link_libraries(CURL::libcurl)
+    endif()
+  else()
+    set(ENABLE_HTTP OFF)
+    message(WARNING "cURL not found, skipping ENABLE_HTTP...")
```

Comment:
why not fail? i know its optional but if its on and it doesnt work better to tell the person hey install curl then continue with the build